docs/contributing/featureflags.md

```diff
@@ -0,0 +1,129 @@
+  provided when requesting the state of a flag. We currently allow targeting on user and
+  organization. Only the IDs are sent to LaunchDarkly to avoid PII sharing.
+- All available feature flag states are provided to clients calling the
+  [configuration API](https://github.com/bitwarden/server/blob/master/src/Api/Models/Response/ConfigResponseModel.cs).
+- Environments (production, QA, and development for now) exist to segment flag states further. This
+  will be automatic based on where code is running.
+
+## Using feature flags in code
+
+Default to an "off" state whenever possible – code defensively so that existing functionality is
+maintained should a flag be unavailable altogether. When an interface supports it, also provide
+default values implying "off" to feature flag accessors.
+
+Offline mode makes default values even more important, and local development as well as self-hosted
+installations imply being offline. Set a safe default value not just in the flag definition online
+at LaunchDarkly but in code.
+
+### Server
+
+1. Inject `IFeatureService` where you need a feature flag. Note that you’ll also need
+   `ICurrentContext` when accessing the feature state.
+2. Check the
+   [FeatureFlagKeys](https://github.com/bitwarden/server/blob/master/src/Core/Constants.cs) constant
+   list for the key you plan on using, and add it if not found.
+3. Utilize the above key constant with the appropriate method on the feature service:
+
+- `IsEnabled` for Booleans, with `false` an assumed default.
+- `GetIntVariation` for integers, with `0` an assumed default.
+- `GetStringVariation` for strings, with `null` an assumed default.
+
+#### File Fallback
+
+For engineers as well as other local users – and this does apply to self-hosted installations if
+they choose to do so, although it is not officially supported at this time – you can use a local
+JSON file to set feature states. Create a `flags.json` file (the name and path can be changed in
+configuration too via `FlagDataFilePath`) with:
+
+```json
+{
+  "flagValues": {
+    "somekey": true
+  }
+}
+```
+
+and set your keys and values. Place the file in the build output directory of what you’re planning
+on running and the system will pick it up; ensure the file is there before starting. It also
+supports live reloading so you can change the file contents and see immediate results in running /
+debugging code.
+
+### Clients / Mobile
+
+:::note
+
+Clients should now refresh configuration upon startup, login, when their local configuration is
+updated, and when sync events come in.
+
+:::
+
+Given the variety of clients and unique implementations, a few general rules:
+
+- Utilize the collection of feature states returned from the configuration API.
+- When in need of a feature state depend on what’s cached locally and available – this should
+  refresh frequently enough.
+- Maintain constants for feature flag keys.
```

Comment:
Not sure I entirely understand your question, but this would be a manual conversion, yes. I am not sure I am aware of the existing flags out there, but one would want to not only implement the config API (if not already) to provide authentication info (a token) in the request so we get targeting support, but to start using the new collection returned for all features. You'll set feature names and keys in LaunchDarkly and refer to the key as a constant everywhere.